### PR TITLE
Chore: Creates new intermediate model int_server_hosting_type 

### PR DESCRIPTION
**Pull request by sri-byte**
Impact: Creates a new intermediate model `int_server_hosting_type` so the logic for hosting type can be reused downstream in other dim and fact models.

### CODE REVIEW

**Review by ifoukarakis — CHANGES_REQUESTED** [review]
`dim_server_info` on the branch contains info where `hosting_type` is null. This is not happening on the master branch, so a bug must have been introduced.

Also checked if there are differences in the hosting type between old and new approach. Only 3 servers seem to have a different hosting type.

File: transform/mattermost-analytics/models/intermediate/product/servers/int_server_hosting_type.sql [hunk 1]
```diff
@@ -11,11 +11,12 @@ with server_telemetry_summary as (
         server_id
 ), server_info as (
    select
-       server_id as server_id,
+       coalesce(st.server_id, ut.server_id) as server_id,
        coalesce(st.count_is_cloud_days, 0) as count_is_cloud_days,
        coalesce(st.count_not_is_cloud_days, 0) as count_not_is_cloud_days
     from
-        server_telemetry_summary st
+        server_telemetry_summary st full outer join
```
Comment (catalintomai) [inline transform/mattermost-analytics/models/intermediate/product/servers/int_server_hosting_type.sql:18]: nit: move "full outer join" statement on next row (in front of the "ref"), to be consistent across the codebase.

Comment (sri-byte) [reply transform/mattermost-analytics/models/intermediate/product/servers/int_server_hosting_type.sql]: Updated.

**Comment by sri-byte** [comment]
> `dim_server_info` on the branch contains info where `hosting_type` is null. This is not happening on the master branch, so a bug must have been introduced.
> 
> Also checked if there are differences in the hosting type between old and new approach. Only 3 servers seem to have a different hosting type.

This is now updated, I missed a join to the user telemetry data which is now added.

**Comment by sri-byte** [comment]
The results now align with master

**Review by catalintomai — APPROVED** [review]

**Review by ifoukarakis — APPROVED** [review]

**Pull request merged**